tests/Integration/Behaviour/Features/Scenario/Shipment/shipment.feature

```diff
@@ -0,0 +1,39 @@
+# ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s shipment
+@restore-all-tables-before-feature
+Feature: Retrieving shipment for orders
+  As a BO users
+  I want to retrieve the list of shipments associated with a specific order
+  In order to be able to track the shipment of this order
+
+  Background:
+    Given I enable feature flag "improved_shipment"
+    And the current currency is "USD"
+    And country "US" is enabled
+    And the module "dummy_payment" is installed
+    And I am logged in as "test@prestashop.com" employee
+    And there is customer "testCustomer" with email "pub@prestashop.com"
+    And customer "testCustomer" has address in "US" country
+    And a carrier "default_carrier" with name "My carrier" exists
+    And I create an empty cart "dummy_cart" for customer "testCustomer"
+    And I select "US" address as delivery and invoice address for customer "testCustomer" in cart "dummy_cart"
+    And I add 1 products "Mug The best is yet to come" to the cart "dummy_cart"
+    And I add 2 products "Mug Today is a good day" to the cart "dummy_cart"
+    And I add order "bo_order1" with the following details:
+      | cart                | dummy_cart                 |
+      | message             | test                       |
+      | payment module name | dummy_payment              |
+      | status              | Awaiting bank wire payment |
+    And I reference order "bo_order1" delivery address as "US"
+
+  Scenario: Retrieve shipmets for existing order
+    Given the order "bo_order1" should have the following shipments:
+      | shipment | shipment1 |
+      | id_carrier | default_carrier |
+      | tracking_number |  |
+      | id_address | US |
+      | shipping_cost_tax_excl | 7.0 |
+      | shipping_cost_tax_incl | 7.42 |
```

Comment:
This should be able to handle a list of shipments so you'll need a horizontal table instead (or vertical it depends on the point of view 😅)
```suggestion
      | shipment | carrier | tracking_number | address | shipping_cost_tax_excl | shipping_cost_tax_incl |
      | shipment1 | default_carrier | | US | 7.0 | 7.42 |
```

So each value will have to be checked in the loop in the Context class, like the carrier ID cannot be fetched outside of the loop since each shipment could have a different carrier
This way the step is already ready for future improvements with multiple shipments